### PR TITLE
created a guide for searching logs

### PR DESCRIPTION
**Pull request by GemTay**
## Purpose
I created a little guide on searching the logs, with a few handy filter snippets

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: docs/runbooks/logs/searching_logs.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#Guide for searching through logs in AWS
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:1]:
```suggestion
# Guide for searching through logs in AWS
```

File: docs/runbooks/logs/searching_logs.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+To begin searching log data, go to CloudWatch --> Logs --> Insights
+
+At the top of the page, select the log group that you want to perform queries on
+
+Enter the queries below and click "Run query"
+
+The order of filters does matter!
+
+#Useful filtering snippets
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:14]:
```suggestion
## Useful filtering snippets
```

File: docs/runbooks/logs/searching_logs.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+```
+| filter @logStream like /(?i)(actor-app)/
+```
+
+Parsing the log message into data categories
+
+```
+| parse @message '* - [*] "* * *" * * *' as ip, datetime, httpverb, path, protocol, responsecode, bytes, logmessage
+```
+Having parsed the data, we can now filter the logs by those categories, such as respond code shown below:
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:25]:
```suggestion

Having parsed the data, we can now filter the logs by those categories, such as respond code shown below:
```

File: docs/runbooks/logs/searching_logs.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+```
+| filter responsecode like /(?i)(4\d\d)/
+```
+Displaying the fields that you want to see:
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:30]:
```suggestion

Displaying the fields that you want to see:
```

File: docs/runbooks/logs/searching_logs.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+```
+| DISPLAY datetime, responsecode, logmessage
+```
+If you want to find log data for a particular user that has reported an issue such as not being able to add their LPA, you can filter for certain log messages found in the code, for example:
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:34]:
```suggestion

If you want to find log data for a particular user that has reported an issue such as not being able to add their LPA, you can filter for certain log messages found in the code, for example:
```

File: docs/runbooks/logs/searching_logs.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+```
+| filter @message like /(?i)(Validating code <code> is inactive)/
+```
+Feel free to add to this list if there are any other filters that you have found useful :)
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:38]:
```suggestion

Feel free to add to this list if there are any other filters that you have found useful :)
```

File: docs/runbooks/logs/searching_logs.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+To begin searching log data, go to CloudWatch --> Logs --> Insights
+
+At the top of the page, select the log group that you want to perform queries on
+
+Enter the queries below and click "Run query"
+
+The order of filters does matter!
+
+#Useful filtering snippets
+Specific log streams eg actor-app, viewer etc
```
Comment (andrewpearce-digital) [inline docs/runbooks/logs/searching_logs.md:15]:
```suggestion
Specific log streams eg actor-app, viewer etc

```

**Comment by GemTay** [comment]
@andrewpearce-digital I've made the changes other than specifying the syntax for the code blocks as I too dont know what to set it as - is it necessary to specify a syntax in this case where I just want the snippets formatted as a code block so that users can copy them?

**Comment by andrewpearce-digital** [comment]
> @andrewpearce-digital I've made the changes other than specifying the syntax for the code blocks as I too don't know what to set it as - is it necessary to specify a syntax in this case where I just want the snippets formatted as a code block so that users can copy them?

Yeah, I had a look and couldn't find it either. In the markdown files AWS uses for their docs pages it's also not specified.

It's not absolutely necessary, just advice from a linter.

**Comment by cooperaj** [comment]
I tend to use "shell script"
As thats what my linter recommended.

```
    ```shell script
        $ #stuff on a shell
    ```
```

**Pull request merged**